### DISCUSSION TITLE
Database first approach

### DISCUSSION THREAD

**Discussion by jstachera** [sqlalchemy/alembic#836] [Usage Questions]
Is there any way to generate model classes from the existing database (in .net core ef this is called scaffolding).
I was not able to find any documentation about that.
When i try to use this command:
`alembic revision --autogenerate -m "Base revision"`
It will generate the migration with downgrade and upgrade method but taking the empty model class as base:
`from sqlalchemy.ext.declarative import declarative_base
Base = declarative_base()
`
Thus, the migration script upgrade contains drop columns from the database and downgrade create.
My intention is to have database as a source and output in the form of generated classes for each table from the database.

Is there anything like that available in alembic?

**Comment by zzzeek** [comment]
hey there -

there is an example of this usage here:

https://alembic.sqlalchemy.org/en/latest/cookbook.html#print-python-code-to-generate-particular-database-tables

note also that you can reflect an entire `MetaData` structure at once for the above recipe using [metadata.reflect](https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=metadata%20reflect#sqlalchemy.schema.MetaData.reflect)

you could also achieve this within the autogenerate process itself, by using `MetaData` with reflect() as the "target_metadata" against a database engine connected to the existing schema, and then in run_migrations_online() you would connect to a blank database.

There's some other ways to do this too, but that first one should get you most of what you need.
